tests/handlers.js

```diff
@@ -17,6 +18,57 @@ const api = path => {
     return `https://api.groupme.com/v3${path}`
 }
 
+function wrapJsonBody(status, body, errors) {
+    const jsonBody = {
+        meta: {
+            code: status,
+        },
+        response: body
+    }
+    if(errors !== undefined) jsonBody.meta.errors = errors
+    
+    return jsonBody
+}
+
+function constructHandler(reqType, path, status, body, bodyType = 'json', errors, headers) {
+    const requestResolver = async (req, res, ctx) => {
+        const resParams = []
+        resParams.push(ctx.status(status))
+        if(body !== undefined) {
+            let bodyContent = body
+            if(typeof body === 'function') bodyContent = body(req)
+            switch(bodyType) {
+                case 'json':
+                    resParams.push(ctx.json(wrapJsonBody(status, bodyContent, errors)))
+                    break
+                case 'text':
+                    resParams.push(ctx.text(bodyContent))
+                    break
+                case 'raw':
+                    resParams.push(ctx.body(bodyContent))
+                    break
+                default:
+                    throw new Error(`Body type not implemented - ${bodyType}`)
+            }
+        }
+        if (headers !== undefined) resParams.push(ctx.set(headers))
+
+        return res(...resParams)
```

Comment:
I don't know the expected syntax of `res()` but for the sake of readability wouldn't it be better to do something like this?
```suggestion
        return res(resStatus, resBody, resHeaders)
```
Also this way, if (in another version of this function down the line) something didn't get set for some reason, the other parameters wouldn't get thrown off by relative indexing of `push()`.

Reply:
It might be better for readability, but there's no need to worry about relative indexing, afaik. All parameters to `res()` are transformers that return functions of the form `(res) => {...}`.